htdocs/product/stock/class/api_productlots.class.php

```diff
@@ -0,0 +1,443 @@
+			throw new RestException(403, "Insufficient rights to read an event");
+		}
+		if ($id === 0) {
+			$result = $this->productlot->initAsSpecimen();
+		} else {
+			$result = $this->productlot->fetch($id);
+			if ($result) {
+				$this->productlot->fetch_optionals();
+				$this->productlot->fetchObjectLinked();
+			}
+		}
+		if (!$result) {
+			throw new RestException(404, 'Product lots not found');
+		}
+
+		return $this->_cleanObjectDatas($this->productlot);
+	}
+
+	/**
+	 * List of product lot
+	 *
+	 * Get a list of product lot
+	 *
+	 * @since	5.0.0	Initial implementation
+	 * @since	21.0.0	Added data pagination
+	 *
+	 * @param	string	$sortfield			Sort field
+	 * @param	string	$sortorder			Sort order
+	 * @param	int		$limit				Limit for list
+	 * @param	int		$page				Page number
+	 * @param	string	$user_ids			User ids filter field (owners of event). Example: '1' or '1,2,3'          {@pattern /^[0-9,]*$/i}
+	 * @param	string	$sqlfilters			Other criteria to filter answers separated by a comma. Syntax example "(pl.label:like:'%dol%') and (pl.datec:<:'20160101')"
+	 * @param	string	$properties			Restrict the data returned to these properties. Ignored if empty. Comma separated list of properties names
+	 * @param	bool	$pagination_data	If this parameter is set to true the response will include pagination data. Default value is false. Page starts from 0*
+	 * @return	array						Array of order objects
+	 * @phan-return productlot[]|array{data:productlot[],pagination:array{total:int,page:int,page_count:int,limit:int}}
+	 * @phpstan-return productlot[]|array{data:productlot[],pagination:array{total:int,page:int,page_count:int,limit:int}}
+	 * @url GET
+	 *
+	 * @throws RestException
+	 */
+	public function index($sortfield = "pl.batch", $sortorder = 'ASC', $limit = 100, $page = 0, $user_ids = '', $sqlfilters = '', $properties = '', $pagination_data = false)
+	{
+		global $conf;
+
+		$obj_ret = [];
+
+		if (!DolibarrApiAccess::$user->hasRight('produit', 'lire')) {
+			throw new RestException(403, "Insufficient rights to view your products lots");
+		}
+
+		$sortorder = (strtoupper($sortorder) === 'DESC') ? 'DESC' : 'ASC';
+
+		$sql  = "SELECT pl.rowid";
+		$sql .= " FROM ".MAIN_DB_PREFIX."product_lot AS pl";
+		$sql .= " LEFT JOIN ".MAIN_DB_PREFIX."product_lot_extrafields AS ple ON ple.fk_object = pl.rowid";
+		$sql .= " WHERE 1=1";
+
+		// Filtres universels
+		if ($sqlfilters) {
+			$errormessage = '';
+			$sql .= forgeSQLFromUniversalSearchCriteria($sqlfilters, $errormessage);
+			if ($errormessage) {
+				throw new RestException(400, 'Error when validating parameter sqlfilters -> '.$errormessage);
+			}
+		}
+
+		// Clone pour total (avant ORDER/LIMIT)
+		$sqlTotals = preg_replace('/^\s*SELECT\s+pl\.rowid/i', 'SELECT COUNT(pl.rowid) AS total', $sql);
+
+		// ORDER BY
+		$sql .= $this->db->order($sortfield, $sortorder);
+
+		// LIMIT/OFFSET
+		if ($limit) {
+			$page   = max(0, (int)$page);
```

Comment:
Not that most IDE should have an option to have this autofixed when you click on save. Check options of your IDE.